web-registry/src/components/organisms/BuyCreditsModal/BuyCreditsModal.tsx

```diff
@@ -81,20 +89,62 @@ const BuyCreditsModal: React.FC<BuyCreditsModalProps> = ({
 }) => {
   const styles = useBuyCreditsModalStyles();
   const theme = useTheme();
-  const walletContext = useWallet();
+  const [selectedSellOrder, setSelectedSellOrder] =
+    useState<ISellOrderInfo | null>(null);
 
   const submit = async (values: BuyCreditsValues): Promise<void> => {
-    const recipient = 'regen18hj7m3skrsrr8lfvwqh66r7zruzdvp6ylwxrx4'; // test account
-    const amount = values.creditCount;
-    if (onSubmit) {
-      onSubmit(values);
-    } else if (walletContext.signSend && walletContext.broadcast) {
-      const txBytes = await walletContext.signSend(amount, recipient);
-      onTxQueued && onTxQueued(txBytes);
-      onClose();
+    const { country, postalCode, stateProvince, retirementAction } = values;
+
+    if (onSubmit && selectedSellOrder) {
+      const fullValues: BuyCreditsValues = {
+        ...values,
+        price: parseInt(selectedSellOrder.askAmount),
+        batchDenom: selectedSellOrder.batchDenom,
+        sellOrderId: selectedSellOrder.id,
+        country: retirementAction === 'autoretire' ? country : '',
+        postalCode: retirementAction === 'autoretire' ? postalCode : '',
+        stateProvince: retirementAction === 'autoretire' ? stateProvince : '',
+      };
+      onSubmit(fullValues);
     }
   };
```

Comment:
Extracting this to `BuyCredits.utils.ts` could make the file more readable.
It could also make the function reusable elsewhere.

Reply:
done